### PR TITLE
Fix edge case in GridFTP Authz where '/' is used

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
# PR Description

Fixes bug in parsing path of globus collection in authz conf file.

# Tasks

* [x] - A description of the PR has been provided, and a diagram included if it is a new feature.
* [ ] - Formatter has been run
* [ ] - CHANGELOG comment has been added
* [ ] - Labels have been assigned to the pr
* [ ] - A reviwer has been added
* [ ] - A user has been assigned to work on the pr
* [ ] - If new feature a unit test has been added

## Summary by Sourcery

Fix an edge case in GridFTP authorization when the root path is used.

Bug Fixes:
- Handle the edge case where the globus_collection_path is set to the root path "/".

Tests:
- Add unit tests for the root path edge case in GridFTP authorization.

### CODE REVIEW

**Comment by JoshuaSBrown** [comment]
@par-hermes format

File: repository/gridftp/globus5/authz/tests/unit/test_AuthzWorker.cpp [hunk 1]
```diff
@@ -211,6 +223,18 @@ BOOST_AUTO_TEST_CASE(URLWithoutHostnameTest) {
...
+BOOST_AUTO_TEST_CASE(RemoveOriginGlobusBaseSetToRoot) {
+  // Test a valid full FTP path when the globus_collection_path is /
+  SDMS::LogContext log_context;
+  config.globus_collection_path[0] = '/';
+  config.globus_collection_path[1] = '\0';
+  SDMS::AuthzWorker worker(&config, log_context);
+  char deep_path[] = "ftp://hostname/globus/root/a/b/c/d/e.txt";
+  std::string expected_result = "/globus/root/a/b/c/d/e.txt";
+
+  BOOST_CHECK_EQUAL(worker.removeOrigin(deep_path), expected_result);
```
Comment (t-ramz) [inline repository/gridftp/globus5/authz/tests/unit/test_AuthzWorker.cpp:235]: What is the difference in output between this and `getAuthzPath`?

**Review by t-ramz — APPROVED** [review]
Looks good, not sure what the big difference is between the two methods

File: repository/gridftp/globus5/authz/tests/unit/test_AuthzWorker.cpp (hunk 1 above)
Comment (JoshuaSBrown) [reply repository/gridftp/globus5/authz/tests/unit/test_AuthzWorker.cpp:235]: The output should be the same, but getAuth method does some additional things. Unit testing here would help identify where the problem is.
